### ISSUE TITLE
Builder with CDATA doesn't work !

### ISSUE THREAD

**Issue by brunovuftv2018** [Leonidas-from-XIV/node-xml2js#535]
Hi guys,

I have a problem on the builder with CDATA. Here is just a simple example on ReadMe:

`var xml2js = require('xml2js');

var obj = {root: {$: {id: "my id"}, **_: "my inner text"**}};

var builder = new xml2js.Builder({ cdata: true });
var xml = builder.buildObject(obj);`

It generates xml:

`<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<root id="my id">my inner text</root>`

**So it misses CDATA wrapping "my inner text"** 

I see in the code of node-xml2js, because of this line: https://github.com/Leonidas-from-XIV/node-xml2js/blob/aefc64af9e9badbd0aade00814548f2ea33c9b4e/src/builder.coffee#L7

`requiresCDATA = (entry) ->
  return typeof entry is "string" && (entry.indexOf('&') >= 0 || entry.indexOf('>') >= 0 || entry.indexOf('<') >= 0)`

Anyone can tell me the reason of this condition, please ?  I think we don't need that

Thank you

**Comment by TanninOne** [comment]
Yes please! I have to generate xml for a foreign application and they will throw an error if certain fields aren't wrapped in CDATA.
So either remove the condition or have the cdata flag as a tristate option like _true | false | 'force'_ where force overrides the condition, that way you stay backwards compatible.

**Comment by brunovuftv2018** [comment]
I fixed that on this repository: https://github.com/brunovuftv2018/node-xml2js

**Comment by Samicelus** [comment]
> Yes please! I have to generate xml for a foreign application and they will throw an error if certain fields aren't wrapped in CDATA.
> So either remove the condition or have the cdata flag as a tristate option like _true | false | 'force'_ where force overrides the condition, that way you stay backwards compatible.

yes, I met the same problem. Please remove this condition or perhaps add anothor option named forceCdata?

**Comment by wmui** [comment]
Why does this issue still exist, this judgment is really redundant, I had to write my own json to xml conversion function



> cdata (default: false): Enclose text nodes with <![CDATA[ ... ]]> when necessary instead of escaping. If not needed, do not add <![CDATA[ ... ]]>.
